src/sciline/pipeline.py

```diff
@@ -39,8 +55,12 @@ class AmbiguousProvider(Exception):
     """Raised when multiple providers are found for a type."""
 
 
-Provider = Callable[..., Any]
-Key = type
+def _indexed_key(index_name: Any, i: int, value_name: Type[T] | Item[T]) -> Item[T]:
```

Comment:
I thought you want to support python 3.8? This use of the pipe was added in 3.10

Reply:
Hmm, odd, why is it passing builds then?